api/src/convert/tx_in.rs

```diff
@@ -87,9 +100,48 @@ impl TryFrom<&external::InputRules> for InputRules {
             .map(tx::TxOut::try_from)
             .collect::<Result<Vec<_>, _>>()?;
         let max_tombstone_block = source.max_tombstone_block;
+        let fractional_outputs = source
+            .fractional_outputs
+            .iter()
+            .map(RevealedTxOut::try_from)
+            .collect::<Result<Vec<_>, _>>()?;
+        let fractional_change = source
+            .fractional_change
+            .as_ref()
+            .map(RevealedTxOut::try_from)
+            .transpose()?;
+        let max_allowed_change_value = source.max_allowed_change_value;
         Ok(InputRules {
             required_outputs,
             max_tombstone_block,
+            fractional_outputs,
+            fractional_change,
+            max_allowed_change_value,
+        })
+    }
+}
+
+/// Convert RevealedTxOut --> external::RevealedTxOut.
+impl From<&RevealedTxOut> for external::RevealedTxOut {
+    fn from(source: &RevealedTxOut) -> Self {
+        let mut result = external::RevealedTxOut::new();
+        result.set_tx_out(external::TxOut::from(&source.tx_out));
+        result.set_amount_shared_secret(source.amount_shared_secret.to_vec());
+        result
+    }
+}
+
+/// Convert external::RevealedTxOut --> RevealedTxOut
+impl TryFrom<&external::RevealedTxOut> for RevealedTxOut {
```

Comment:
Please add some tests exercising these conversions, e.g. round-trip conversion tests are useful for catching missing field mappings